sdk/search/azure-search/src/test/java/com/azure/search/LookupSyncTests.java

```diff
@@ -0,0 +1,361 @@
+// Licensed under the MIT License.
+package com.azure.search;
+
+import com.azure.search.models.GeoPoint;
+import com.azure.search.test.environment.models.Hotel;
+import com.azure.search.test.environment.models.HotelAddress;
+import com.azure.search.test.environment.models.HotelRoom;
+import com.azure.search.test.environment.models.ModelWithPrimitiveCollections;
+import org.junit.Assert;
+import org.junit.jupiter.api.Test;
+
+import java.text.ParseException;
+import java.time.OffsetDateTime;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static java.lang.Double.NEGATIVE_INFINITY;
+import static java.lang.Double.NaN;
+import static java.lang.Double.POSITIVE_INFINITY;
+import static org.unitils.reflectionassert.ReflectionAssert.assertReflectionEquals;
+import static org.unitils.reflectionassert.ReflectionComparatorMode.IGNORE_DEFAULTS;
+
+public class LookupSyncTests extends LookupTestBase {
+    private SearchIndexClient client;
+
+    @Test
+    public void canGetStaticallyTypedDocument() throws ParseException {
+        createHotelIndex();
+        client = getSearchIndexClientBuilder(INDEX_NAME).buildClient();
+
+        Hotel expected = prepareExpectedHotel();
+        uploadDocument(client, expected);
+
+        Document result = client.getDocument(expected.hotelId());
+        Hotel actual = convertToType(result, Hotel.class);
+        assertReflectionEquals(expected, actual, IGNORE_DEFAULTS);
+    }
+
+    @Test
+    public void canGetStaticallyTypedDocumentWithNullOrEmptyValues() {
+        createHotelIndex();
+        client = getSearchIndexClientBuilder(INDEX_NAME).buildClient();
+
+        Hotel expected = prepareEmptyHotel();
+        uploadDocument(client, expected);
+
+        Document result = client.getDocument(expected.hotelId());
+        Hotel actual = convertToType(result, Hotel.class);
+        assertReflectionEquals(expected, actual, IGNORE_DEFAULTS);
+    }
+
+    @Test
+    public void canGetStaticallyTypedDocumentWithPascalCaseFields() {
+        createHotelIndex();
+        client = getSearchIndexClientBuilder(INDEX_NAME).buildClient();
+
+        Hotel expected = preparePascalCaseFieldsHotel();
+        uploadDocument(client, expected);
+
+        Document result = client.getDocument(expected.hotelId());
+        Hotel actual = convertToType(result, Hotel.class);
+        assertReflectionEquals(expected, actual, IGNORE_DEFAULTS);
+    }
+
+    @Test
+    public void canRoundtripStaticallyTypedPrimitiveCollections() {
+        String indexName = setupIndexWithDataTypes();
+        client = getSearchIndexClientBuilder(indexName).buildClient();
+
+        ModelWithPrimitiveCollections expected = preparePrimitivesModel();
+        uploadDocument(client, expected);
+
+        Document result = client.getDocument(expected.key());
+        ModelWithPrimitiveCollections actual = convertToType(result, ModelWithPrimitiveCollections.class);
```

Comment:
@brjohnstmsf the tests passed, we do not expsitly convert to any type, but leave the documents as a map of objects. Checking out the test recording, the spefical cases "-INF", "INF", 'NaN" are set there.

Reply:
@navalev The test passing is no guarantee that the behavior is correct, because the test could be flawed (it probably isn't, but I can't explain how it is passing based on the implementation I see). The session records don't tell us anything useful either, because the important aspect of the test is how the objects are represented in memory. They are definitely correct on the wire.

If I understand it correctly, the test is asserting that the `doubles` property of the object returned from `convertToType` actually contains the `Double` values for positive and negative infinity, and not strings or other double values. I'm confused because I can't explain why this test is passing, because I can't find code that converts "INF" to `Double.POSITIVE_INFINITY` and "-INF" to `Double.NEGATIVE_INFINITY`. In .NET, we needed special logic for that because JSON.NET expects "Infinity" and "-Infinity", not "INF" and "-INF", which are specific to OData as far as I know.

I apologize if this seems nit-picky, but we need to be sure that data integrity is maintained in all scenarios, and if it can't, that we understand what those scenarios are so we can document them. These are corner cases that have given us trouble in the past, so I have to get a complete understanding of how they work in the Java SDK.

Reply:
The assertion is done between the input that is created in method preparePrimitivesModel (and inseters the types NEGATIVE_INFINITY, NaN and POSITIVE_INFINITY) and the result of the lookup API (getDocument) which returns a map of <String, Object>. This map is being converted into an object using Jackson objectMapper (convertToType method) and the 2 object are being compred using the assertReflectionEquals method of org.untilis http://www.unitils.org/apidocs/org/unitils/reflectionassert/ReflectionAssert.html
If this does not seem to be sufficient, a task to imporve the test can be added to the maintenance backlog.

Reply:
We will investigate this after the PR is merged.